src/main/java/org/opensearch/security/securityconf/impl/v7/InternalUserV7.java

```diff
@@ -51,6 +51,7 @@ public class InternalUserV7 implements Hideable, Hashed, StaticDefinable {
     private Map<String, String> attributes = Collections.emptyMap();
     private String description;
     private List<String> opendistro_security_roles = Collections.emptyList();
+    private List<String> opensearch_security_roles = Collections.emptyList();
```

Comment:
One high-level remark: 

IMHO, config property names should be as self-descriptive as possible. However, neither `opendistro_security_roles` nor `opensearch_security_roles` are really self-descriptive.

The function of this property is as follows: Roles which are specified via `role` property first go through role-mapping so that they are mapped onto the effective roles of the user at runtime. Roles specified via `opendistro_security_roles`  do not go through role mapping, but get effective immediately.

Can we find a property name that is (kind of) self-descriptive?

Reply:
what do you think about ``opensearch_security_inline_roles``/``opensearch_security_static_roles``/``opensearch_security_direct_roles`` or any other name :) ?

Reply:
I do not think, the `opensearch_security` prefix is necessary, as the other attribute is also just called `roles`.

So, that would be `inline_roles` or `static_roles` or `direct_roles`. I would not recommend `static_roles`, as this could be confused with the static role configuration in https://github.com/opensearch-project/security/blob/main/src/main/resources/static_config/static_roles.yml .

I could also imagine `directly_effective_roles`, that would be most telling.

@cwperks @DarshitChanpura @derek-ho wdyt?

Reply:
Personally, I think we should consider removing this attribute from the InternalUser object and only do roles mappings with RolesMappings.

What is the advantage of having multiple ways to map roles to usernames?

Reply:
> What is the advantage of having multiple ways to map roles to usernames?

I would consider these roles as default permissions that a user has when you create such an user. Also, it's simpler to setup for small number of users. 

Using role mappings is better for more centralized and flexible management, better for managing similar users, etc. 

I think it should be okay to keep this feature for it's simplicity (single API setup/configuration) - also, maybe it will prove helpful with service/system accounts feature in the plugin - i.e. using inline_roles or such directly_effective_roles for accounts.

Reply:
> @cwperks should we create an issue to track the long-term re-evaluation of role mappings.

Sounds like we are going to keep supporting inline roles based on this conversation. I think we can add an issue for migration assistance here. In past versions, there was a Migrate API to help migrating the security config so should we start implementing one to account for these upcoming changes as well?